python/opendataloader-pdf-mcp/src/opendataloader_pdf_mcp/mineru.py

```
@@ -0,0 +1,55 @@
+        if shutil.which("mineru") is None:
+            raise MinerUError("mineru not found in PATH")
+
+        try:
+            result = subprocess.run(
+                ["mineru", "-p", str(pdf_path), "-o", str(output_dir), "--method", "auto"],
+                capture_output=True,
+                text=True,
+                timeout=_MINERU_TIMEOUT,
+            )
+        except subprocess.TimeoutExpired:
+            raise MinerUError(f"mineru timed out after {_MINERU_TIMEOUT}s")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)